android/sdl_android/src/main/java/com/smartdevicelink/proxy/SdlProxyBase.java

```diff
@@ -8061,150 +8061,182 @@ public void onServiceError(SdlSession session, SessionType type, String reason)
+					continue;
+				}
 
-			}
+				TouchCoord touchCoord = touchCoordList.get(touchCoordList.size() - 1);
+				if (touchCoord == null) {
+					continue;
+				}
 
+				int motionEventAction = sdlMotionEvent.getMotionEventAction(touchType, touchEvent);
+				long downTime = sdlMotionEvent.downTime;
+				long eventTime = sdlMotionEvent.eventTime;
+				pointer = sdlMotionEvent.getPointerById(touchEvent.getId());
+				if (pointer != null) {
+					pointer.setCoords(touchCoord.getX() * touchScalar[0], touchCoord.getY() * touchScalar[1]);
+				}
 
-			if(sdlMotionEvent == null) {
-				if (touchType == TouchType.BEGIN) {
-					sdlMotionEvent = new SdlMotionEvent();
-				}else{
-					return  null;
+				MotionEvent.PointerProperties[] pointerProperties;
+				MotionEvent.PointerCoords[] pointerCoords;
+				pointerProperties = new MotionEvent.PointerProperties[sdlMotionEvent.pointers.size()];
+				pointerCoords = new MotionEvent.PointerCoords[sdlMotionEvent.pointers.size()];
+
+				for (int i = 0; i < sdlMotionEvent.pointers.size(); i++) {
+					pointerProperties[i] = new MotionEvent.PointerProperties();
+					pointerProperties[i].id = sdlMotionEvent.getPointerByIndex(i).id;
+					pointerProperties[i].toolType = MotionEvent.TOOL_TYPE_FINGER;
+
+					pointerCoords[i] = new MotionEvent.PointerCoords();
+					pointerCoords[i].x = sdlMotionEvent.getPointerByIndex(i).x;
+					pointerCoords[i].y = sdlMotionEvent.getPointerByIndex(i).y;
+					pointerCoords[i].orientation = 0;
+					pointerCoords[i].pressure = 1.0f;
+					pointerCoords[i].size = 1;
 				}
-			}
 
-			int eventAction = sdlMotionEvent.getMotionEvent(touchType, pointerProperties);
-			long startTime = sdlMotionEvent.startOfEvent;
+				motionEvent = MotionEvent.obtain(downTime, eventTime, motionEventAction,
+						sdlMotionEvent.pointers.size(), pointerProperties, pointerCoords, 0, 0, 1,
+						1, 0, 0, InputDevice.SOURCE_TOUCHSCREEN, 0);
+				motionEventList.add(motionEvent);
 
-			//If the motion event should be finished we should clear our reference
-			if(eventAction == MotionEvent.ACTION_UP || eventAction == MotionEvent.ACTION_CANCEL){
-				sdlMotionEvent = null;
+				if (motionEventAction == MotionEvent.ACTION_UP || motionEventAction == MotionEvent.ACTION_CANCEL) {
+					//If the motion event should be finished we should clear our reference
+					sdlMotionEvent = null;
+					break;
+				} else if ((motionEventAction & MotionEvent.ACTION_MASK) == MotionEvent.ACTION_POINTER_UP) {
+					sdlMotionEvent.removePointerById(touchEvent.getId());
+				}
 			}
 
-			return MotionEvent.obtain(startTime, SystemClock.uptimeMillis(), eventAction, eventListSize, pointerProperties, pointerCoords, 0, 0,1,1,0,0, InputDevice.SOURCE_TOUCHSCREEN,0);
-
+			return motionEventList;
 		}
-
-
-
 	}
 
 	/**
 	 * Keeps track of the current motion event for VPM
 	 */
-	private static class SdlMotionEvent{
-		long startOfEvent;
-		SparseIntArray pointerStatuses = new SparseIntArray();
+	private static class SdlMotionEvent {

```

Comment:
Can you move the method `onTransportUpdate` above this class and the `convertTouchEvent` so that this class is at the end of the file?